var/spack/repos/builtin/packages/openfoam-org/50-etc.patch

```diff
@@ -0,0 +1,24 @@
+--- OpenFOAM-5.x.org/etc/bashrc	2017-07-25 18:43:40.000000000 +0200
++++ OpenFOAM-5.x/etc/bashrc	2017-12-11 13:36:09.479818186 +0100
+@@ -42,15 +42,17 @@
+ #
+ # Please set to the appropriate path if the default is not correct.
+ #
+-[ ${BASH_SOURCE:-$0} ] && \
+-export FOAM_INST_DIR=$(cd $(dirname ${BASH_SOURCE:-$0})/../.. && pwd -P) || \
+-export FOAM_INST_DIR=$HOME/$WM_PROJECT
+-# export FOAM_INST_DIR=~$WM_PROJECT
++rc="${BASH_SOURCE:-${ZSH_NAME:+$0}}"
++[ -n "$rc" ] && FOAM_INST_DIR=$(\cd $(dirname $rc)/../.. && \pwd -P) || \
```

Comment:
See previous comment about why it is wrong to combine the `export ..` with the logical.
Additionally, we have a backslash escape on the 'cd' to ensure that we get the real builtin `cd` and not a shell function with pushd/popd. Remembering that at this stage we are sourcing, not executing the bashrc file. I had reports of this tripping up a few users.
Of course, we cannot safeguard against everything, but for this somewhat critical statement it seems like a good idea.

Reply:
Also, some clusters have their installations on different disks and use symlinks for logical organization. Thus in most cases it is actually much more useful to have `pwd -L` (see the openfoam-com version) instead of `pwd -P` , but I left this as is since I don't wish to force this opinion on openfoam.org fork.